scripts/auto-rebase/rebase.py

```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import logging
+import argparse
+import subprocess
+from collections import namedtuple
+
+from git import Repo, PushInfo  # GitPython
+from github import GithubIntegration, Github  # pygithub
+from pathlib import Path
+
+APP_ID_ENV = "APP_ID"
+KEY_ENV = "KEY"
+ORG_ENV = "ORG"
+REPO_ENV = "REPO"
+AMD64_RELEASE_ENV = "AMD64_RELEASE"
+ARM64_RELEASE_ENV = "ARM64_RELEASE"
+JOB_NAME_ENV = "JOB_NAME"
+BUILD_ID_ENV = "BUILD_ID"
+DRY_RUN_ENV = "DRY_RUN"
+
+BOT_REMOTE_NAME = "bot-creds"
+REMOTE_ORIGIN = "origin"
+
+# List of reviewers to always requestes review from
+REVIEWERS = ["pmtk", "ggiguash"]
+
+# If True, then just log action such as branch push and PR or comment creation
+REMOTE_DRY_RUN = False
+
+_extra_msgs = []
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s %(levelname)s %(message)s')
+
+
+def try_get_env(var_name, die=True):
+    val = os.getenv(var_name)
+    if val is None or val == "":
+        if die:
+            logging.error(f"Could not get environment variable '{var_name}'")
+            sys.exit(f"Could not get environment variable '{var_name}'")
+        else:
+            logging.info(f"Could not get environment variable '{var_name}' - ignoring")
+            return ""
+    return val
+
+
+def run_rebase_sh(release_amd64, release_arm64):
+    script_dir = os.path.abspath(os.path.dirname(__file__))
+    args = [f"{script_dir}/rebase.sh", "to", release_amd64, release_arm64]
+    logging.info(f"Running: '{' '.join(args)}'")
+    result = subprocess.run(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True)
+    logging.info(f"Output:\n\n{result.stdout}\n------------------------------\n")
+    logging.info(f"Script returned code: {result.returncode}")
+    rr = namedtuple("RebaseResult", ["success", "output"])
```

Comment:
This type really should be defined outside of the function so it's clear it's used elsewhere.

Reply:
Done